app/app.js

```diff
@@ -368,30 +380,58 @@ var AppComponent = React.createClass({
 
   showLogin: function() {
     this.renderPage = this.renderLogin;
+    //always check
+    this.finialiseSignup();
```

Comment:
Can you change these to American spelling - `finalizeSignup`? It's what we use elsewhere (e.g., 'authorize')